testing/logging/test_fixture.py

```diff
@@ -172,6 +172,30 @@ def test_caplog_captures_for_all_stages(caplog, logging_during_setup_and_teardow
     assert set(caplog._item.stash[caplog_records_key]) == {"setup", "call"}
 
 
+def test_caplog_filters_by_logger(
+    pytester: Pytester, caplog, logging_during_setup_and_teardown
+):
+    """Tests that records are filtered properly by the given logger name."""
+    assert not caplog.records
+    caplog.set_level("INFO")
+    logger.info("a_call_log")
+    pytester.makepyfile(
```

Comment:
I don't understand why you need to use pytester in your test. I think it suffices if you only uses the `caplog` fixture and not the `pytester` fixture.